### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by ybjozee**
This PR does the following:
1. Update the Symfony dependencies to version 7
2. Removes the FOSUserBundle as it is no longer maintained
3. Replaces the *cspoo/swiftmailer-mailgun-bundle* with the native mailer Symfony component (alongside the mailgun integration. 
4. Removes the web-server dependency as it is no longer maintained. It is recommended that the project be served locally via the Symfony CLI
5. Bumps the minimum PHP version from PHP 7.1 to 8.2
6. Uses rector to migrate from annotations to attributes

This PR does not fix the breaking changes that occur in the code as the result of the breaking changes. These will be fixed in a subsequent PR.

### CODE REVIEW

File: rector.php
Comment (yemiwebby) [inline rector.php:1]: Out of curiosity, what does this file do?

**Review by yemiwebby — APPROVED** [review]
Weldone man. Top work 💯

File: rector.php
Comment (ybjozee) [reply rector.php:1]: I used [Rector](https://packagist.org/packages/rector/rector) to automate the refactoring of the PHP code. The *rector.php* file contains the rules for the automation.

Comment (yemiwebby) [reply rector.php:1]: Cool

**Pull request merged**